R/hazardFunctions.R

```diff
@@ -20,23 +20,46 @@ ExpHazOS <- function(t, h01, h02, h12) {
   ((h12 - h02) * (h01 + h02) - h01 * h12 * exp(-h012 * t)) / ((h12 - h02) - h01 * exp(-h012 * t))
 }
 
+#' Helper Function for `avgHRExpOS()`
+#'
+#' @param x (`numeric`)\cr variable of integration.
+#' @param h01 (positive `number`)\cr transition hazard for 0 to 1 transition.
+#' @param h02 (positive `number`)\cr transition hazard for 0 to 2 transition.
+#' @param h12 (positive `number`)\cr transition hazard for 1 to 2 transition.
+#' @param h0 (`list`)\cr transition parameters for the first treatment group.
+#' @param h1 (`list`)\cr transition parameters for the second treatment group.
+#' @param alpha (`number`)\cr weight parameter, see `avgHRExpOS`.
+#' @return This returns the value of the integrand used to calculate
+#' the average hazard ratio for constant transition hazards, see `avgHRExpOS()`.
+#' @export
+#'
+#' @examples
+#' h0 <- list(h01 = 0.18, h02 = 0.06, h12 = 0.17)
+#' h1 <- list(h01 = 0.23, h02 = 0.07, h12 = 0.19)
+#' avgHRInteg(x = 5, h01 = 0.2, h02 = 0.5, h12 = 0.7, h0 = h0, h1 = h1, alpha = 0.5)
+avgHRInteg <- function(x, h01, h02, h12, h0, h1, alpha) {
```

Comment:
here I would at least add an assertion for `alpha` because that is used in the calculation directly below. the other arguments are only forwarded to other functions, so we don't need to assert them here